src/options/views/tab-settings/vm-export.vue

```diff
@@ -137,60 +114,57 @@ function normalizeFilename(name) {
     values: withValues,
-  })
-  .then((data) => {
-    const names = {};
-    const vm = {
-      scripts: {},
-      settings: options.get(),
+  });
+  const names = {};
+  const vm = {
+    scripts: {},
+    settings: options.get(),
+  };
+  delete vm.settings.sync;
+  if (withValues) vm.values = {};
+  const files = (objectGet(data, 'items') || []).map(({ script, code }) => {
+    let name = normalizeFilename(getScriptName(script));
+    if (names[name]) {
+      names[name] += 1;
+      name = `${name}_${names[name]}`;
+    } else names[name] = 1;
+    const { lastModified, lastUpdated } = script.props;
+    const info = {
+      custom: script.custom,
+      config: script.config,
+      position: script.props.position,
+      lastModified,
+      lastUpdated,
     };
-    delete vm.settings.sync;
-    if (withValues) vm.values = {};
-    const files = (objectGet(data, 'items') || []).map(({ script, code }) => {
-      let name = normalizeFilename(getScriptName(script));
-      if (names[name]) {
-        names[name] += 1;
-        name = `${name}_${names[name]}`;
-      } else names[name] = 1;
-      const { lastModified, lastUpdated } = script.props;
-      const info = {
-        custom: script.custom,
-        config: script.config,
-        position: script.props.position,
-        lastModified,
-        lastUpdated,
-      };
-      if (withValues) {
-        // `values` are related to scripts by `props.id` in Violentmonkey,
-        // but by the global `props.uri` when exported.
-        const values = data.values[script.props.id];
-        if (values) vm.values[script.props.uri] = values;
-      }
-      vm.scripts[name] = info;
-      return {
-        name: `${name}.user.js`,
-        content: code,
-        lastModDate: new Date(lastUpdated || lastModified),
-      };
-    });
-    files.push({
-      name: 'violentmonkey',
-      content: JSON.stringify(vm),
-    });
-    return files;
-  })
-  .then(files => files.reduce((result, file) => (
-    result.then(writer => addFile(writer, file))
-  ), getWriter()))
-  .then(writer => new Promise((resolve) => {
-    writer.close((blob) => {
-      resolve(blob);
-    });
-  }));
+    if (withValues) {
+      // `values` are related to scripts by `props.id` in Violentmonkey,
+      // but by the global `props.uri` when exported.
+      const values = data.values[script.props.id];
+      if (values) vm.values[script.props.uri] = values;
+    }
+    vm.scripts[name] = info;
+    return {
+      name: `${name}.user.js`,
+      content: code,
+      lastModDate: new Date(lastUpdated || lastModified),
+    };
+  });
+  files.push({
+    name: 'violentmonkey',
+    content: JSON.stringify(vm),
+  });
+  const zip = await loadZip();
+  const blobWriter = new zip.BlobWriter('application/zip');
+  const writer = new zip.ZipWriter(blobWriter);
+  await Promise.all(files.map(file => writer.add(file.name, new zip.TextReader(file.content), {
+    lastModDate: file.lastModDate,
```

Comment:
Yeah it's fast now but I wonder why [the documentation](https://gildas-lormeau.github.io/zip.js/core-api.html#zip-writing) suggests `bufferedWrite: true`? The output is identical in both cases.

Reply:
The author has given a detailed clarification [here](https://github.com/gildas-lormeau/zip.js/issues/236).
I think it does not matter much in our case, but may make a difference when files are large.